stdlib/Test/docs/src/index.md

```diff
@@ -332,6 +332,148 @@ Test.detect_ambiguities
 Test.detect_unbound_args
 ```
 
+## Workflow for Testing Packages
+
+Using the tools available to us in the previous sections, here is a potential workflow of creating a package and adding tests to it.
+
+### Generating an Example Package
+
+For this workflow, we will create a package called `Example`:
+
+```julia
+pkg> generate Example
+shell> cd Example
+shell> mkdir test
+pkg> activate .
+```
+
+### Creating Sample Functions
+
+The number one requirement for testing a package is to have functionality to test.
+For that, we will add some simple functions to `Example` that we can test.
+Add the following to `src/Example.jl`:
+
+```julia
+module Example
+
+function greet()
+    "Hello world!"
+end
+
+function simple_add(a, b)
+    a + b
+end
+
+function type_multiply(a::Float64, b::Float64)
+    a * b
+end
+
+end
+```
+
+### Creating a Test Environment
+
+From within the root of the `Example` package, navigate to the `test` directory, activate a new environment there, and add the `Test` package to the environment:
+
+```julia
+shell> cd test
+pkg> activate .
+(test) pkg> add Test
+```
+
+### Testing Our Package
+
+Now, we are ready to add tests to `Example`.
+It is standard practice to create a file within the `test` directory called `runtests.jl` which contains the test sets we want to run.
+Go ahead and create that file within the `test` directory and add the following code to it:
+
+```julia
+using Example
+using Test
+
+@testset "Example tests" begin 
+
+	@testset "Math tests" begin
```

Comment:
You can leave the `begin` out on master here:

```
shell> cat test.jl
foo() = @test true

@testset foo()

julia> using Test

julia> @testset "My Tests" include("test.jl")
Test Summary: | Pass  Total  Time
My Tests      |    1      1  0.3s
Test.DefaultTestSet("My Tests", Any[Test.DefaultTestSet("foo", Any[], 1, false, false, true, 1.660635314428429e9, 1.660635314582263e9, false)], 0, false, false, true, 1.660635314317867e9, 1.660635314582275e9, false)
```

Reply:
Good point. As 1.8 has just been released, I think to afford backwards compatibility, I may leave in the `begin` syntax. Does that seem reasonable to you @seelengrab?

Reply:
Good point. As 1.8 has just been released, I think to afford backwards compatibility, I may leave in the `begin` syntax. Does that seem reasonable to you @seelengrab?

Reply:
The docs probably won't be backported, so there shouldn't be any harm in changing it - I'm ambivalent either way :shrug: The idea was to have an example using that pattern in the docs and not just the docstring, to show people that they can group their tests like that much more easily.